Makefile

```diff
@@ -31,6 +31,8 @@ TEST_OBJ_PATH = $(TEST_BIN_PATH)/obj
 CFLAGS += -I$(SRC_PATH) -I$(SRC_PATH)/wrappers/themis/ -I/usr/local/include -fPIC
 LDFLAGS += -L/usr/local/lib
 
+unexport CFLAGS LDFLAGS
```

Comment:
we shouldn't have a makefile with side-effects (unexpected behavior for users).

if `unexport` is required for CircleCI tests only, should wrap it using appropriate flag.

Reply:
Honestly, I'd personally call the default makefile behavior unexpected.

I was under impression that make variables get their initial values from the environment and command-line, but after that they are independent. And if the makefile explicitly wants to export some variable to spawned subprocesses then it should

```make
export SOME_VARIABLE
```

However, it turns out that if the user does

```
SOME_VARIABLE=value make
```

then make _will_ export the variable, but if the variable is not set then it won't (regardless of whether SOME_VARIABLE is assigned by the makefile itself or not).

I think that with this `unexport` clause the behavior will be reasonable and expected for the users:

- By default `CFLAGS` is not set and Themis will be built with the default set of flags as specified by Themis makefile. Any other activities called by make will use their own defaults and will not be affected by flags set for Themis.

- The user can set _environment variable_ `CFLAGS` to the initial set of flags for the compiler.

  Like so:
  ```
  export CFLAGS="-I/some/important/path"
  make install
  ```
  or so:
  ```
  CFLAGS="-I/some/important/path" make install
  ```

  Themis binaries will be built with these flags **_plus_** the additional flags added by Themis makefile (our warnings, include directories, etc.) Our additional flags affect only Themis binaries. If Themis makefile calls some other process then its CFLAGS environment variable will _not_ contain Themis custom flags.

  For example, package maintainers may be interested in this feature. Someone who wants to cross-compile Themis will likely need it as well.

  Without `unexport` in this case the subprocesses spawned by make will have their CFLAGS set to the initial environment value _plus_ all the Themis custom flags which is not what they probably expect.

- Finally, the user can pass `CFLAGS` on the make _command line_ like this:

  ```
  make CFLAGS="-Wmy-personal-flags"
  ```

  This forces the specified flags to be used for building Themis. Flags added by our makefile will be ignored. Make has this feature for power users who _need_ a specific set of flags.

  (We can override this override if necessary with `override`.)

(Ditto for LDFLAGS.)